### PR TITLE
948 - Fixed toggle indeterminate activateRow index with datagrid

### LINKED ISSUE #948: activateRow and deactivateRow toggle state incorrectly
**Describe the bug**
For indeterminate paging toggleRowActivation changes activated index across pages.

Per Tim:
- move logic from toggleRow to either activateRow or deactivateRow
- change toggleRow to check state and call either activateRow or deactivateRow

**To Reproduce**
1. Go to http://master-enterprise.demo.design.infor.com/components/datagrid/example-paging-indeterminate-landmark.html
2. Click on 2nd row (but not the checkbox) to show it activated in light blue
3. Click next page paging bar button
4. 2nd row index on 2nd page isn't activated. Also, Trying to click on it to activate it is failing as well. Have to click on a nother row and then back to get 2nd row index to activate.

**Expected behavior**
For indeterminate paging 2nd row index on 2nd page and any subsequent page should be activated.

**Platform**
 - Device: PC
 - OS Version: Windows 10
 - Browser Name: Chrome
 - Browser Version 69.0.3497.100 (Official Build) (64-bit)

### PR DESCRIPTION
**Pull request by deep7102**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
For indeterminate paging activated index was not working across pages.

**Related github/jira issue (required)**:
Closes #948

**Steps necessary to review your pull request (required)**:
http://localhost:4000/components/datagrid/example-paging-indeterminate-landmark.html
- Open above link
- Click on 2nd row (but not the checkbox) to show it activated in light blue
- Click next page paging bar button
- 2nd row on 2nd page and any subsequent page should be activated

### CODE REVIEW

File: src/components/datagrid/datagrid.js [hunk 1]
```diff
@@ -5885,69 +5885,149 @@ Datagrid.prototype = {
...
-      }
+  /**
+   * Activate given row with mixed selection mode.
+   * @private
+   * @param  {object} row The row to activated
+   * @param  {number} idx The row index to activated
+   * @param  {object} dataset Optional data to use
+   * @returns {void}
+   */
+  activateMixSelRow(row, idx, dataset) {
```
Comment (tmcconechy) [inline src/components/datagrid/datagrid.js]: Would prefer activateMixedSelectionRow (even tho long its kind of our standard)

File: src/components/datagrid/datagrid.js [hunk 2]
```diff
@@ -5885,69 +5885,149 @@ Datagrid.prototype = {
...
+
+  /**
+  * Deactivate given row with mixed selection mode.
+  * @private
+  * @param  {object} row The row to deactivated
+  * @param  {number} idx The row index to deactivated
+  * @param  {object} dataset Optional data to use
+  * @returns {void}
+  */
+  deactivateMixSelRow(row, idx, dataset) {
```
Comment (tmcconechy) [inline src/components/datagrid/datagrid.js]: Would prefer deactivateMixedSelectionRow (even tho long its kind of our standard)

File: src/components/datagrid/datagrid.js [hunk 3]
```diff
@@ -5885,69 +5885,149 @@ Datagrid.prototype = {
...
+    }
+  },
+
+  /**
+  * Deactivate all rows with mixed selection mode.
+  * @private
+  * @param  {object} dataset Optional data to use
+  * @returns {void}
+  */
+  deactivateAllMixSelRows(dataset) {
```
Comment (tmcconechy) [inline src/components/datagrid/datagrid.js]: Would prefer deactivateAllMixedSelectionRows (even tho long its kind of our standard)

**Comment by tmcconechy** [comment]
Just needs a few tests, please.

**Comment by deep7102** [comment]
Updated with functional and e2e tests now

**Comment by deep7102** [comment]
Updated now, for mixed selectable tree grid. While clicking rows downwards from Id 1 to Id 10,
 then click upwards. Clicked row was not activating.

**Pull request merged**